.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedDateTimeExtractor.cs

```diff
@@ -35,17 +35,35 @@ public static bool HasTokenIndex(string text, Regex regex, out int index)
             return false;
         }
 
-        public static bool TryMergeModifierToken(ExtractResult er, Regex tokenRegex, string text)
+        public bool TryMergeModifierToken(ExtractResult er, Regex tokenRegex, string text, bool ambiguous = false)
         {
             var beforeStr = text.Substring(0, er.Start ?? 0);
 
+            // Avoid adding mod for "from" in "from ... to ..."
+            if (ambiguous && this.config.AmbiguousRangeModifierPrefix.IsMatch(beforeStr))
+            {
+                var matches = this.config.PotentialAmbiguousRangeRegex.Matches(text).Cast<Match>();
+                if (matches.Any(m => m.Index < er.Start + er.Length && m.Index + m.Length > er.Start))
+                {
+                    return false;
+                }
+            }
+
             if (HasTokenIndex(beforeStr.TrimEnd(), tokenRegex, out var tokenIndex))
             {
                 var modLength = beforeStr.Length - tokenIndex;
 
                 er.Length += modLength;
                 er.Start -= modLength;
                 er.Text = text.Substring(er.Start ?? 0, er.Length ?? 0);
+                if (er.Metadata == null)
```

Comment:
This if/else could be a simple method AssignModMetadata() or something like that. It would cleanup the current method a little and make it easier to change metadata in the future.